x-pack/plugins/ml/public/alerting/preview_alert_condition.tsx

```diff
@@ -0,0 +1,277 @@
+      },
+      {
+        title: i18n.translate('xpack.ml.previewAlert.scoreLabel', {
+          defaultMessage: 'Anomaly score:',
+        }),
+        description: score,
+      },
+      ...(topInfluencers && topInfluencers.length > 0
+        ? [
+            {
+              title: i18n.translate('xpack.ml.previewAlert.topInfluencersLabel', {
+                defaultMessage: 'Top influencers:',
+              }),
+              description: (
+                <ul>
+                  {topInfluencers.map((i) => (
+                    <li key={i.unique_key}>
+                      <EuiCode transparentBackground>{i.influencer_field_name}</EuiCode> ={' '}
+                      {i.influencer_field_value} [{i.score}]
+                    </li>
+                  ))}
+                </ul>
+              ),
+            },
+          ]
+        : []),
+      ...(topRecords && topRecords.length > 0
+        ? [
+            {
+              title: i18n.translate('xpack.ml.previewAlert.topRecordsLabel', {
+                defaultMessage: 'Top records:',
+              }),
+              description: (
+                <ul>
+                  {topRecords.map((i) => (
+                    <li key={i.unique_key}>
+                      <EuiCode transparentBackground>
+                        {i.function}({i.field_name})
+                      </EuiCode>{' '}
+                      {i.by_field_value} {i.over_field_value} {i.partition_field_value} [{i.score}]
+                    </li>
+                  ))}
+                </ul>
+              ),
+            },
+          ]
+        : []),
+    ];
+
+    return <EuiDescriptionList type={'row'} compressed={true} listItems={listItems} />;
+  }
+);
+
+export const PreviewAlertCondition: FC<PreviewAlertConditionProps> = ({
+  alertingApiService,
+  alertParams,
+}) => {
+  const [lookBehindInterval, setLookBehindInterval] = useState<string>();
+  const [areResultsVisible, setAreResultVisible] = useState<boolean>(true);
+  const [previewError, setPreviewError] = useState<Error | undefined>();
+  const [previewResponse, setPreviewResponse] = useState<PreviewResponse | undefined>();
+
+  const validators = useMemo(
+    () => composeValidators(requiredValidator(), timeIntervalInputValidator()),
+    []
+  );
+
+  const validationErrors = useMemo(() => validators(lookBehindInterval), [lookBehindInterval]);
+
+  useEffect(
+    function resetPreview() {
+      setPreviewResponse(undefined);
+    },
+    [alertParams]
+  );
+
+  const testCondition = useCallback(async () => {
+    try {
+      const response = await alertingApiService.preview({
+        alertParams,
+        timeRange: lookBehindInterval!,
+      });
+      setPreviewResponse(response);
+      setPreviewError(undefined);
+    } catch (e) {
+      setPreviewResponse(undefined);
+      setPreviewError(e.body ?? e);
+    }
+  }, [alertParams, lookBehindInterval]);
+
+  /**
+   * Sample of the 5 results from the top hits
```

Comment:
When there are more than 5 results, I think you should add a label to the results preview saying something like `Showing 5 examples`, or maybe at the bottom `and 6 others`

![image](https://user-images.githubusercontent.com/7405507/107358229-0c2b4600-6acb-11eb-9f0f-316d46ad0f68.png)

Reply:
Done in efa96c68bf032b10a46e7a204a8ae7982885a6f2.
I also added `sampleSize` parameter for the preview endpoint to limit the number of docs we send to the client